### DISCUSSION TITLE
tqk_get 실행 시 에러 발생 문의 : lexical error: invalid char in json text.

### DISCUSSION THREAD

**Discussion by earth2u** [mrchypark/tqk#50] [Q&A]
안녕하세요.

인터넷 검색하면서 한국주식을 조회할수 있는 R 패키지가 있다는 사실을 알게 되어서 제공된 문서와 같이 실행하였는데 계속 동일한 에러가 발생하고 구글에 검색해도 JSON 파싱에 대한 내용만 있어서 문의드리고자 합니다.
다른 유저분들은 이상없이 사용하는 걸로 보여서 PC상에 패키지 설치 등 문제가 아닐까 싶은데 해결방법을 아시는지 싶어서요.

install.packages("tqk")
library(tqk)
tqk_get("095570")
에러: lexical error: invalid char in json text.
                                  	  		<html>  <head>    <title>에러
                     (right here) ------^

![image](https://github.com/mrchypark/tqk/assets/175316907/200a20bd-1c7f-4696-a954-a63375df9116)

**Comment by mrchypark** [comment]
안녕하세요 tqk_get() 함수가 의존하는 사이트 구성이 변경된 것 같습니다. 확인해보고 다시 댓글 드릴게요. 이슈 작성해주셔서 감사합니다.

**Reply by mrchypark** [reply]
확인해본 결과 현재 사이트가 문제가 있다고 나오네요. 아래 링크가 동작할 때 사용해주시고, 그래도 문제가 있으면 알려주시면 감사하겠습니다. 패키지에는 위 문제를 안내하는 방법을 추가해야겠네요. 

http://data.krx.co.kr/comm/bldAttendant/getJsonData.cmd
![image](https://github.com/mrchypark/tqk/assets/6179259/e20d7873-1210-4ed2-8f99-86041be2929f)

**Comment by earth2u** [comment]
빠른 회신 감사합니다.

저는 거제에 있는 조선소 it 부서에 근무중으로
재무부서에 제공하기 위해 공공 api 쪽 활용을 검토하다가
개발하신 내용을 인터넷으로 접하게 되었습니다.
(R tidyquant, fredr 활용 미국 경제지표 활용)

한국에도 r 능력자가 있어서 대단하단 생각을 했습니다.
KRX 사이트 정상화 될때 다시 실행해보겠습니다.

2024년 7월 11일 목요일, ChanYub Park ***@***.***>님이 작성:

**Comment by earth2u** [comment]
안녕하세요. 
KRX사이트에 오픈API Q&A에 알려주신 URL 관련 질의를 올렸는데 담당자로부터 DB접근용 URL로 웹을 통해서는 접근이 안된다고 답변을 받았습니다. 확인요청드립니다.

1. KRX 질의 사이트 : http://openapi.krx.co.kr/contents/OPP/COMM/qna/OPPCOMM003_S1.cmd
2. 답변내용 : 
안녕하십니까, KRX정보데이터시스템에 대한 귀하의 관심에 감사드립니다.
문의주신 URL은 KRX 정보데이터시스템내 데이터 표출을 위한 DB 접속 URL 입니다.
웹페이지를 통해 해당 URL에 접속하려는 시도는 비정상적인 접근 요청입니다.
정상적인 루트로 KRX 정보데이터시스템 접근시 이용 가능합니다.
감사합니다.
![image](https://github.com/user-attachments/assets/c1067469-1e54-4bc0-a098-8803c3504222)

**Reply by mrchypark** [reply]
확인했습니다. 경로가 막힌거군요. 다른 접근이 가능한지 확인해보겠습니다. 정보 공유해주셔서 감사합니다.
